### PR TITLE
Sample 12 prevent reconnection and move to functional component

### LINKED ISSUE #2609: Minimizable sample should not reconnect
Direct line reconnects to the bot framework after minimiz and maximiz the bot window in the browser 

- The impact : 
 The user will receive a welcome message after every time he changes the window size

- Test environment:
 1- Download the [minimizable web chat example](https://github.com/microsoft/BotFramework-WebChat/tree/master/samples/12.customization-minimizable-web-chat) 
 2- Then update web chat from 4.5.1 to 4.6.0
 3- After reciving the welcome message from the bot click on minimz button then click on maximiz button

- Please find below the screenshot attached for reference.
  First connection 
2019-11-18_15h40_47

 After minimiz and maximiz the window 

2019-11-18_15h40_02
 

Web chat version 4.6.0

### LINKED ISSUE #2750: Discuss: Who should call directLine.connect() and directLine.disconnect()
## Feature Request

Today, we call `directLine.connect()` when the React component is up. Technically, `<Composer>` will call `useEffect` to dispatch a connect action to saga. And call dispatch a disconnect when `<Composer>` is being unmounted.

### Minimizable scenario

This affect minimizable scenario. In minimizable scenario, we want the Web Chat UI to be complete/almost gone from DOM tree to save resources. While the DOM component is gone, we want to maintain the connection.

### Should `core` owns the connection, or `component`?

Today, `<Composer>` owns the connection. Should it be `createStore` to own it?

## Proposals

Please list your proposals below.

### A new option for the ownership of DirectLine connection

If the ownership is set to outside of Web Chat, Web Chat will not call `connect/disconnect` and would assume it is connected.

[Enhancement]

### PR DESCRIPTION
**Pull request by compulim**
> Fixes #2609.

### Fixed

-  Fixes [#2609](https://github.com/microsoft/BotFramework-WebChat/issues/2609). Fix sample 12: minimizable button is causing another reconnect on restore, by [@compulim](https://github.com/compulim) in PR [#2758](https://github.com/microsoft/BotFramework-WebChat/pull/2758)

## Description

In sample 12 "minimizable Web Chat", when Web Chat is minimized, it disconnect Direct Line. When Web Chat is restored, it reconnect.

Temporarily, we hide Web Chat by setting `display: none`. This is not an ideal fix.

We need #2750 to be resolved before we could implement an ideal fix.

## Specific Changes

- Modify sample 12 to use functional components
- When minimized/restored, we will set `display: none` instead of unmounting Web Chat from DOM tree

---

-  [x] ~Testing Added~
   -  No tests for samples currently

### CODE REVIEW

**Comment by corinagum** [comment]
> We need to resolve #2750 before we could implement an ideal fix.

This text in your PR description will close issue 2750, so you should remove the resolve in front of it.

File: samples/12.customization-minimizable-web-chat/src/MinimizableWebChat.js [hunk 1]
```diff
@@ -1,116 +1,112 @@
...
-            payload: {
-              name: 'webchat/join',
-              value: {
-                language: window.navigator.language
+const MinimizableWebChat = () => {
+  const store = useMemo(
+    () =>
+      createStore({}, ({ dispatch }) => next => action => {
+        if (action.type === 'DIRECT_LINE/CONNECT_FULFILLED') {
+          setTimeout(() => {
```
Comment (tdurnford) [inline samples/12.customization-minimizable-web-chat/src/MinimizableWebChat.js]: Why do we have this in a timeout?

Comment (compulim) [reply samples/12.customization-minimizable-web-chat/src/MinimizableWebChat.js]: Yes, checked we don't need that. I was naively converting the code. 😆

**Comment by corinagum** [comment]
Is there a reason this hasn't been merged in? We have a ton of potentially stale PRs.

**Pull request merged**